tests/test_load_data.py

```diff
@@ -27,271 +37,239 @@ class TestLoadData(SMRFTestCase):
+
+        # test the base run with the config file
+        result = self.can_i_run_smrf(config)
+        assert result
+
+
+class TestLoadGrid(SMRFTestCase):
 
     def test_grid_wrf(self):
         """ WRF NetCDF loading """
 
         config = deepcopy(self.base_config)
         del config.raw_cfg['csv']
 
-        wrf_grid = {'data_type': 'wrf',
-                    'file': './RME/gridded/WRF_test.nc',
-                    'zone_number': 11,
-                    'zone_letter': 'N'}
-        config.raw_cfg['gridded'] = wrf_grid
-#         config.raw_cfg['system']['max_values'] = 2
-        config.raw_cfg['system']['threading'] = False
-#         config.raw_cfg['system']['timeout'] = 10
+        adj_config = {'gridded':{'data_type': 'wrf',
+                                 'wrf_file': './RME/gridded/WRF_test.nc',
+                                 'zone_number': '11',
+                                 'zone_letter': 'N'},
+                      'system':{'threading':'False',
+                                'log_file': './output/log.txt'},
+                     'precip':{'station_adjust_for_undercatch': 'False'},
+                     'thermal':{'correct_cloud': 'False',
+                                'correct_veg': 'True'},
+                     'time':{'start_date':'2015-03-03 00:00',
+                             'end_date': '2015-03-03 04:00'}}
 
-        # set the distrition to grid, thermal defaults will be fine
-        variables = ['air_temp', 'vapor_pressure', 'wind', 'precip', 'solar', 'thermal']
-        for v in variables:
-             config.raw_cfg[v]['mask'] = False
+        config.raw_cfg.update(adj_config)
+
+        # set the distribution to grid, thermal defaults will be fine
+        for v in self.dist_variables:
+             config.raw_cfg[v]['grid_mask'] = 'False'
 
-        config.raw_cfg['precip']['adjust_for_undercatch'] = False
-        config.raw_cfg['thermal']['correct_cloud'] = False
-        config.raw_cfg['thermal']['correct_veg'] = True
 
         # fix the time to that of the WRF_test.nc
-        config.raw_cfg['time']['start_date'] = '2015-03-03 00:00'
-        config.raw_cfg['time']['end_date'] = '2015-03-03 04:00'
         config.apply_recipes()
         config = cast_all_variables(config, config.mcfg)
 
-
         # ensure that the recipes are used
-        self.assertTrue(config.raw_cfg['precip']['adjust_for_undercatch'] == False)
-        self.assertTrue(config.raw_cfg['thermal']['correct_cloud'] == False)
-        self.assertTrue(config.raw_cfg['thermal']['correct_veg'] == True)
+        assert 'station_adjust_for_undercatch' not in config.cfg['precip'].keys()
+        self.assertTrue(config.cfg['thermal']['correct_cloud'] == False)
+        self.assertTrue(config.cfg['thermal']['correct_veg'] == True)
 
         result = can_i_run_smrf(config)
         self.assertTrue(result)
 
+    @unittest.skip(" Skipping over hrrr tests for issues with eccodes")
```

Comment:
This makes me uncomfortable since HRRR is now how we run real time. Has this been tested with a machine that has eccodes or can we try/except on an eccodes error?

Reply:
I think I did and I think I recall it working. I will try again in the docker to double check. If it does pass, I can just leave it uncommented and we can add in the try and except.

Reply:
Running the test in Docker will bypass this.

Reply:
Found an issue with test_load_grid_data. TestLoadGrid. test_grid_hrrr and other hrrr tests. 

L588 in model_framework.py  was 
```python 
if self.distribute['thermal'].gridded and \
   self.config['gridded']['data_type'] != 'hrrr':
    self.distribute['thermal'].distribute_thermal(self.data.thermal.loc[t],
                                                              self.distribute['air_temp'].air_temp)
```

Since we changed the names here and restructured them a bit this was attempting to distribute thermal data  for  data_type == hrrr_grib or hrrr_netcdf. It sounds like this is only for when the data_type == wrf. So I have changed the above code to:

```python 
if self.distribute['thermal'].gridded and \
   self.config['gridded']['data_type'] == 'wrf':
    self.distribute['thermal'].distribute_thermal(self.data.thermal.loc[t],
                                                              self.distribute['air_temp'].air_temp)
```
Our current types are ['hrrr_grib','hrrr_netcdf', 'wrf', 'netcdf'] 

**So this solution will assume that  ['hrrr_grib','hrrr_netcdf',  'netcdf']  don't need to distribute thermal.**

@scotthavens  please confirm this is what we want here.

Reply:
On second pass `wrf` and `netcdf` distribute thermal, `hrrr_grib` and `hrrr_netcdf` do not distribute thermal in this manner.

Reply:
Ok then I will make it:

```python 
if self.distribute['thermal'].gridded and \
   self.config['gridded']['data_type'] in ['wrf','netcdf']:
    self.distribute['thermal'].distribute_thermal(self.data.thermal.loc[t],
                                                              self.distribute['air_temp'].air_temp)
```